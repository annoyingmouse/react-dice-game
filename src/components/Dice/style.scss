.view{
  width:200px;
  height:200px;
  margin:auto;
  perspective:600px;
  margin: 2em 0;
  .dice{
    width:200px;
    height:200px;
    position:absolute;
    transform-style:preserve-3d;
    transition: transform 1s;
    &[data-value="1"]{
      transform: rotateY(360deg) !important; 
    }
    &[data-value="2"]{
      transform: rotateY(-90deg) !important;
    }
    &[data-value="3"]{
      transform: rotateX(-90deg) !important;
    }
    &[data-value="4"]{
      transform: rotateY(90deg) !important;
    }
    &[data-value="5"]{
      transform: rotateX(90deg) !important;
    }
    &[data-value="6"]{
      transform: rotateY(180deg) !important;
    }
    .side{
      position: absolute;
      width: 200px;
      height: 200px;
      background: #fff;
      box-shadow: inset 0 0 40px #ccc;
      border-radius: 40px;
      &.inner {
        background: #e0e0e0;
        box-shadow: none;
      }
      &.front {
        transform:translateZ(100px);
        &.inner {
          transform: translateZ(98px);
        }
      }
      &.right {
        transform:rotateY(90deg) translateZ(100px);
        &.inner {
          transform: rotateY(90deg) translateZ(98px);
        }
      }
      &.back{
        transform:rotateY(180deg) translateZ(100px);
        &.inner{
          transform: rotateX(-180deg) translateZ(98px);
        }
      }
      &.left{
        transform:rotateY(-90deg) translateZ(100px);
        &.inner{
          transform: rotateY(-90deg) translateZ(98px);
        }
      }
      &.top{
        transform:rotateX(90deg) translateZ(100px);
        &.inner{
          transform: rotateX(90deg) translateZ(98px);
        }
      }
      &.bottom{
        transform:rotateX(-90deg) translateZ(100px);
        &.inner{
          transform: rotateX(-90deg) translateZ(98px);
        }
      }
      .dot {
        position:absolute;
        width:46px;
        height:46px;
        border-radius:23px;
        background:#444;
        box-shadow:inset 5px 0 10px #000;
        &.center {
          margin:77px 0 0 77px;
        }
        &.dtop {
          margin-top:20px;
        }
        &.dleft {
          margin-left:134px;
        }
        &.dright {
          margin-left:20px;
        }
        &.dbottom {
          margin-top:134px;
        }
        &.center.dleft {
          margin:77px 0 0 20px;
        }
        &.center.dright {
          margin:77px 0 0 134px;
        }
      }
    }
  }
}
@keyframes shake {
  0% { transform: translate(1px, 1px) rotate(0deg); }
  10% { transform: translate(-1px, -2px) rotate(-1deg); }
  20% { transform: translate(-3px, 0px) rotate(1deg); }
  30% { transform: translate(3px, 2px) rotate(0deg); }
  40% { transform: translate(1px, -1px) rotate(1deg); }
  50% { transform: translate(-1px, 2px) rotate(-1deg); }
  60% { transform: translate(-3px, 1px) rotate(0deg); }
  70% { transform: translate(3px, 1px) rotate(-1deg); }
  80% { transform: translate(-1px, -1px) rotate(1deg); }
  90% { transform: translate(1px, 2px) rotate(0deg); }
  100% { transform: translate(1px, -2px) rotate(-1deg); }
}