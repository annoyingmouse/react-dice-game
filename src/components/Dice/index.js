import React from 'react';
import './style.scss';

export const Dice = props => {
  return (
    <section className={props.player.decorator}>
      <div className="container">
        <h2>{props.player.name}</h2>
        <div className="view">
          <div className="dice" data-value={props.player.value}>
            <div className="side front">
              <div className="dot center"></div>
            </div>
            <div className="side front inner"></div>
            <div className="side right">
              <div className="dot dtop dleft"></div>
              <div className="dot dbottom dright"></div>
            </div>
            <div className="side top inner"></div>
            <div className="side back">
              <div className="dot dtop dleft"></div>
              <div className="dot dtop dright"></div>
              <div className="dot dbottom dleft"></div>
              <div className="dot dbottom dright"></div>
              <div className="dot center dleft"></div>
              <div className="dot center dright"></div>
            </div>
            <div className="side back inner"></div>
            <div className="side left">
              <div className="dot dtop dleft"></div>
              <div className="dot dtop dright"></div>
              <div className="dot dbottom dleft"></div>
              <div className="dot dbottom dright"></div>
            </div>
            <div className="side left inner"></div>
            <div className="side top">
              <div className="dot dtop dleft"></div>
              <div className="dot center"></div>
              <div className="dot dbottom dright"></div>
            </div>
            <div className="side top inner"></div>
            <div className="side bottom">
              <div className="dot center"></div>
              <div className="dot dtop dleft"></div>
              <div className="dot dtop dright"></div>
              <div className="dot dbottom dleft"></div>
              <div className="dot dbottom dright"></div>
            </div>
            <div className="side bottom inner"></div>
          </div>
        </div>
        <h3>Wins {props.player.score}</h3>
      </div>
    </section>
  )
}
