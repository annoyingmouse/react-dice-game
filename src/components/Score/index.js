import React from 'react';
import './style.scss'
import { connect } from 'react-redux'

const Score = props => {
  return (
    <section className="score">
      <div className="container">
        <h1>Let's go!</h1>
        <button
          onClick={props.roll}
          disabled={props.disabled}
          id="roll">
          Roll Dice!
        </button>
        <h3>{props.state.message}</h3>
      </div>
    </section>
  )
}

export default connect(state => ({
  state
}))(Score)