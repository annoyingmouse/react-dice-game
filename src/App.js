import React, { Component } from 'react';
import './style.scss';
import { Dice } from './components/Dice/'
import { Score } from './components/Score/'

class App extends Component {
  constructor() {
    super()
    this.state = {
      disabled: false,
      games: 0,
      player: {
        name: 'You',
        score: 0,
        decorator: 'player',
        value: 1
      },
      computer: {
        name: 'Computer',
        score: 0,
        decorator: 'computer',
        value: 1
      }
    }
    this.rollDice = this.rollDice.bind(this)
    this.updateDimensions = this.updateDimensions.bind(this);
  }
  componentDidUpdate (prevProps, prevState) {
    if (this.state.disabled) {
      this.enableButton = setTimeout(() => {
        this.setState((state) => {
          const original = { ...state }
          original.disabled = false
          return original
        })
      }, 1000)
    }
  }
  componentWillUnmount () {
    clearTimeout(this.enableButton);
    window.removeEventListener("resize", this.updateDimensions);
  }
  updateDimensions () {
    const windowWidth = window.innerWidth / 2
    const windowHeight = window.innerHeight / 2
    document.querySelectorAll(".container").forEach(element => {
      console.log()
      if(400 > windowWidth || 400 > windowHeight){
        const scale = Math.min(
          windowWidth / 400,    
          windowHeight / 400
        );
        element.style.transform = `translate(-50%, -50%) scale(${scale})`
      }
    })
  }
  componentDidMount () {
    this.updateDimensions()
    window.addEventListener("resize", this.updateDimensions);
  }
  rollDice () {
    if (!this.state.disabled) {
      this.setState(state => {
        const original = { ...state }
        original.player.value = Math.floor(Math.random() * 6) + 1
        original.computer.value = Math.floor(Math.random() * 6) + 1
        original.disabled = true;
        if (original.player.value < original.computer.value) {
          original.computer.score = state.computer.score + 1
        }
        if (original.player.value > original.computer.value) {
          original.player.score = state.player.score + 1
        }
        original.games = state.games + 1;
        return original
      });
    }
  }
  render () {
    return (
      <div className="app">
        <Dice player={this.state.player} />
        <Dice player={this.state.computer} />
        <Score disabled={this.state.disabled} roll={this.rollDice} />
      </div>
    );
  }
}

export default App;
