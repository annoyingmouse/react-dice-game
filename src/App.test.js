import React from 'react'
import { shallow, mount, render } from 'enzyme';
import App from './App'
import { Dice } from './components/Dice/'


describe('React component test with Enzyme', () => {
  // TODO fix tests now we've moved over from state to Redux
  it('renders without crashing', () => {
    shallow(<App />);
  });
  it('should render correctly', () => {
    const component = shallow(<App />);
    expect(component).toMatchSnapshot();
  });
  it('click event on button', () => {
    const component = mount(<App />);
    component.find('#roll').simulate('click');
    expect(component).toMatchSnapshot();
    component.unmount();
  });
  it('Check state of disabled after mounting', () => {
    const component = mount(<App />);
    expect(component.instance().state.disabled).toBe(false);
    component.unmount();
  });
  it('Check state of disabled after clicking the button', () => {
    const component = mount(<App />);
    component.find('#roll').simulate('click');
    expect(component.instance().state.disabled).toBe(true);
    component.unmount();
  });
  it('Check state of disabled after clicking the button', () => {
    const component = mount(<App />);
    expect(component.instance().state.games).toBe(0);
    component.unmount();
  });
  it('Check state of disabled after clicking the button', () => {
    const component = mount(<App />);
    component.find('#roll').simulate('click');
    expect(component.instance().state.games).toBe(1);
    component.unmount();
  });
});

// describe('Check Dice', () => {
//   it('check there are two dice', () => {
//     const component = mount(<App />);
//     expect(component.find('.dice').length).toBe(2);
//   });
//   it('check there is one', () => {
//     const computer = {
//       name: 'Computer',
//       score: 0,
//       decorator: 'computer',
//       value: 1
//     }
//     const component = shallow(<Dice player={computer} />);
//     console.log("component", component);
//     expect(component.find('.dice[data-value=1]').first()).toBe(1);
//   });
// });