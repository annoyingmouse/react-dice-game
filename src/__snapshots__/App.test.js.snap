// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React component test with Enzyme should render correctly 1`] = `ShallowWrapper {}`;
